lib/fluent/plugin/prometheus.rb

```diff
@@ -85,6 +85,19 @@ def self.placeholder_expander(log)
         Fluent::Plugin::Prometheus::ExpandBuilder.new(log: log)
       end
 
+      def stringify_keys(hash_to_stringify)
+        # Adapted from: https://www.jvt.me/posts/2019/09/07/ruby-hash-keys-string-symbol/
+        h = hash_to_stringify.map do |k,v|
+          v_str = if v.instance_of? Hash
+                    v.stringify_keys
```

Comment:
It doesn't work, `v` has to be passed to `stringify_keys`.

Reply:
:scream:

Reply:
Recursive form can handle nested hash:


```ruby
def stringify_keys(hash_to_stringify)
  # Adapted from: https://www.jvt.me/posts/2019/09/07/ruby-hash-keys-string-symbol/
  arr = hash_to_stringify.map do |k,v|
    v_str = if v.instance_of? Hash
              stringify_keys(v)
            else
              v
            end
    [k.to_s, v_str]
  end
  arr.to_h
end
```

```irb
irb> hash = {:a => 1, :b => { :c => 1, :d => 2, :e=> {:f=> 3}}}
irb> puts stringify_keys(hash)

#=> {"a"=>1, "b"=>{"c"=>1, "d"=>2, "e"=>{"f"=>3}}}
```